xtask/src/main.rs

```diff
@@ -713,3 +725,159 @@ fn host_tests(workspace: &Path, args: HostTestsArgs) -> Result<()> {
 
     Ok(())
 }
+
+/// Find the most recently modified rlib for the specified package in the deps_dir.
+fn find_esp_hal_rlib(deps_dir: &Path, package: &Package) -> Result<PathBuf> {
```

Comment:
I don't think we should find this, we should build the correct configuration and check it. This means building esp-hal as a library _without_ the rt feature enabled.